includes/Checker/Checks/File_Type_Check.php

```diff
@@ -152,13 +164,12 @@ function ( $directory ) use ( $directories ) {
 			// Only use an error in production, otherwise a warning.
 			$is_error = ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) && 'production' === wp_get_environment_type();
 			foreach ( $vcs_directories as $dir ) {
-				$result->add_message(
+				$this->add_result_message_for_file(
+					$result,
 					$is_error,
 					__( 'Version control checkouts should not be present.', 'plugin-check' ),
-					array(
-						'code' => 'vcs_present',
-						'file' => str_replace( $result->plugin()->path(), '', $dir ),
-					)
+					'vcs_present',
+					str_replace( $result->plugin()->path(), '', $dir )
```

Comment:
The `str_replace()` is handled in the central method now and therefore not needed here.
```suggestion
					$dir
```